catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/CatalogMigratableImpl.java

```diff
@@ -0,0 +1,221 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CatalogMigratableImpl.class);
+
+    private static final String DEFAULT_FAILURE_MESSAGE = "Catalog could not export metacards";
+
+    private static final String CATALOG_EXPORT_DIRECTORY = "org.codice.ddf.catalog";
+
+    private final CatalogFramework framework;
+
+    private final FilterBuilder filterBuilder;
+
+    private final MigrationFileWriter fileWriter;
+
+    private final CatalogMigratableConfig config;
+
+    private MigrationTaskManager taskManager;
+
+    /**
+     * Constructor.
+     *
+     * @param description   description of this migratable
+     * @param framework     framework used to retrieve the metacards to export
+     * @param filterBuilder builder used to create query filters
+     * @param fileWriter    object used to write metacards to disk
+     * @param config        export configuration information
+     */
+    public CatalogMigratableImpl(@NotNull String description, @NotNull CatalogFramework framework,
+            @NotNull FilterBuilder filterBuilder, @NotNull MigrationFileWriter fileWriter,
+            @NotNull CatalogMigratableConfig config) {
+        super(description, true);
+
+        notNull(framework, "CatalogFramework cannot be null");
+        notNull(filterBuilder, "FilterBuilder cannot be null");
+        notNull(fileWriter, "FileWriter cannot be null");
+        notNull(config, "Configuration object cannot be null");
+
+        this.framework = framework;
+        this.filterBuilder = filterBuilder;
+        this.fileWriter = fileWriter;
+        this.config = config;
+    }
+
+    /**
+     * Exports all the metacards currently stored in the catalog framework.
+     * <p>
+     * {@inheritDoc}
+     */
+    public MigrationMetadata export(Path exportPath) throws MigrationException {
+        taskManager = createTaskManager(config, createExecutorService(config));
+        config.setExportPath(exportPath.resolve(CATALOG_EXPORT_DIRECTORY));
+        fileWriter.createExportDirectory(config.getExportPath());
+
+        Collection<MigrationWarning> warnings = new ArrayList<>();
+        Map<String, Serializable> props = createMapWithNativeQueryMode();
+        Filter dumpFilter = filterBuilder.attribute(Metacard.ANY_TEXT)
+                .is()
+                .like()
+                .text("*");
+
+        QueryImpl exportQuery = createQuery(dumpFilter);
+        exportQuery.setPageSize(config.getExportQueryPageSize());
+        exportQuery.setRequestsTotalResultsCount(false);
+
+        QueryRequest exportQueryRequest = new QueryRequestImpl(exportQuery, props);
+
+        Subject subject = retrieveSystemSubject();
+        if (subject == null) {
+            LOGGER.error("System Subject was null");
+            throw new ExportMigrationException(DEFAULT_FAILURE_MESSAGE);
+        }
+
+        try {
+            subject.execute(new CatalogMigrationRunner(exportQuery, exportQueryRequest));
```

Comment:
How does that make this code harder to test? If you get rid of the CatalogMigrationRunner, you also remove the need to have the factory method to mock it out.